core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -217,22 +230,22 @@ private URI buildUri(String path, Map<String, String> params) {
    */
   private <T> T execute(
       Method method,
-      String path,
+      String baseUri,
```

Comment:
I don't think we should be changing the parameter names here.  This is still the path, it's just that it can be a full path as opposed to a relative path.  I don't feel `baseUri` makes sense in this context.